File: 704. Binary Search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target)
    {
        return binarySearch(nums,0,nums.size()-1,target);
    }

    int binarySearch(vector<int> nums,int left,int right,int target)
    {
        if(left<=right)
        {
            int mid = (right+left)/2;
            
            if(target==nums[mid])
            {
                return mid;
            }
            if(target>nums[mid])
            {
                return binarySearch(nums,mid+1,right,target);
            }
            if(target<nums[mid])
            {
                return binarySearch(nums,left,mid-1,target);
            }

        }
        return -1;
    }
};

File: 704. Binary Search.py
class Solution(object):
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """

        left = 0
        right = len(nums)-1

        while(left<=right):
            mid = (left+right)/2
            
            if(target==nums[mid]):
                return mid
            if(target<nums[mid]):
                right = mid - 1
                continue
            else:
                left = mid + 1
                continue
                
        return -1